src/server/activeobjectmgr.cpp

```diff
@@ -128,39 +132,40 @@ void ActiveObjectMgr::invalidateActiveObjectObserverCaches()
 	}
 }
 
+void ActiveObjectMgr::updatePos(u16 id, const v3f &pos) {
+	// HACK only update if we already know the object
```

Comment:
This probably isn't that bad of a hack (if the object has not been registered yet, it will be registered later, with the correct position), though ideally positions should only be updated once the object has been registered. Fixing the above hack probably helps with this.